src/server/pachyderm_test.go

```diff
@@ -7883,6 +7883,298 @@ func TestStatsDeleteAll(t *testing.T) {
+	require.NoError(t, err)
+	require.YesError(t, w.Close())
+	commit, err = c.StartCommit(dataRepo, "invalid2")
+	require.NoError(t, err)
+	w, err = c.PutFileSplitWriter(dataRepo, "invalid2", "data", pfs.Delimiter_SQL, 0, 0, false, nil, nil)
+	require.NoError(t, err)
+	_, err = w.Write([]byte(strings.Join(rows, "") + pgDumpFooter))
+	require.NoError(t, err)
+	require.YesError(t, w.Close())
+
+	// Test Valid pgdump
+	commit, err = c.StartCommit(dataRepo, "master")
+	require.NoError(t, err)
+	w, err = c.PutFileSplitWriter(dataRepo, "master", "data", pfs.Delimiter_SQL, 0, 0, false, nil, nil)
+	require.NoError(t, err)
+	fullPGDump := pgDumpHeader + strings.Join(rows, "") + pgDumpFooter
+	_, err = w.Write([]byte(fullPGDump))
+	require.NoError(t, err)
+	require.NoError(t, w.Close())
+	w, err = c.PutFileSplitWriter(dataRepo, "master", "data2", pfs.Delimiter_SQL, 0, 0, false, nil, nil)
+	fullPGDump2 := pgDumpHeader2 + strings.Join(rows2, "") + pgDumpFooter2
+	_, err = w.Write([]byte(fullPGDump2))
+	require.NoError(t, err)
+	require.NoError(t, w.Close())
+	require.NoError(t, c.FinishCommit(dataRepo, commit.ID))
+
+	fileInfos, err := c.ListFile(commit.Repo.Name, commit.ID, "data")
+	require.NoError(t, err)
+	require.Equal(t, 3, len(fileInfos))
+	fileInfos2, err := c.ListFile(commit.Repo.Name, commit.ID, "data2")
+	require.NoError(t, err)
+	require.Equal(t, 2, len(fileInfos2))
+
+	var buf bytes.Buffer
+	for i := 0; i < len(fileInfos); i++ {
+		buf.Reset()
+		require.NoError(t, c.GetFile(commit.Repo.Name, commit.ID, fileInfos[i].File.Path, 0, 0, &buf))
+		expected := pgDumpHeader + rows[i] + pgDumpFooter
+		require.Equal(t, expected, buf.String())
+		require.Equal(t, len(expected), int(fileInfos[i].SizeBytes))
+	}
+	// The dir should only have the header/footer
+	buf.Reset()
+	require.NoError(t, c.GetFile(commit.Repo.Name, commit.ID, "data", 0, 0, &buf))
+	require.Equal(t, pgDumpHeader+pgDumpFooter, buf.String())
+	dirFileInfo, err := c.InspectFile(commit.Repo.Name, commit.ID, "data")
+	require.NoError(t, err)
+	dirExpectedLen := len(pgDumpHeader) + len(pgDumpFooter)
+	for i := 0; i < len(rows); i++ {
+		dirExpectedLen += len(rows[i])
+	}
+	require.Equal(t, dirExpectedLen, int(dirFileInfo.SizeBytes))
+
+	// Validate glob read behavior
+	buf.Reset()
+	require.NoError(t, c.GetFile(commit.Repo.Name, commit.ID, "/data/*", 0, 0, &buf))
+	// glob file is non deterministic ... so we need to check against all permutations of the rows:
+	validDumps := []string{
+		pgDumpHeader + strings.Join(rows, "") + pgDumpFooter,
+		pgDumpHeader + strings.Join([]string{rows[0], rows[2], rows[1]}, "") + pgDumpFooter,
+		pgDumpHeader + strings.Join([]string{rows[1], rows[0], rows[2]}, "") + pgDumpFooter,
+		pgDumpHeader + strings.Join([]string{rows[1], rows[2], rows[0]}, "") + pgDumpFooter,
+		pgDumpHeader + strings.Join([]string{rows[2], rows[0], rows[1]}, "") + pgDumpFooter,
+		pgDumpHeader + strings.Join([]string{rows[2], rows[1], rows[0]}, "") + pgDumpFooter,
+	}
```

Comment:
Doesn't `GetFile` with a glob return things in sorted order? It certainly looks like there's some sorting going on in there.

Reply:
Huh. Yea, it definitely does. I'd seen output w rows out of order during testing, but that must've been a bug in the way I was testing, because it's definitely ordered in the output. 

Anyway, this is much simpler now.